### PR TITLE
Move cast icon to the mini player

### PR DESCRIPTION
**Pull request by ashiagr**
Part of #2081 

## Description
This moves the cast icon to the mini-player.

#### Featuer Flag ON

1. Open the app 
2. ✅ Notice that the cast icon is not visible on the Podcasts and Filters tab
3. Play an episode
4. ✅ Notice that the cast icon is visible on the mini-player
5. Add a few episodes to up next queue
6. ✅ Notice that the up-next icon is still visible on the mini-player and the cast icon is shown adjacent to it.
7. Tap the cast icon on the mini-player
8. ✅ Notice that `chromecast_view_shown` event is tracked as before
9. Smoke test that casting works as before

#### Featuer Flag OFF

1. Turn off `Show Up Next in tab bar` feature flag in Profile -> Settings -> Beta features
2. Kill and restart the app
3. Notice that the cast icon is visible at its original place

#### Featuer Flag ON

Mini player - No upnext | Mini player- With upnext | Filters | Themed
---|---|----|----
0 mini player- ni upnext | 1 upnext | 2 filters| tehemed

#### Featuer Flag OFF

Mini player - No upnext | Mini player- With upnext | Filters
---|---|----
3 1miniplayer-noupnext-ff-diasabled | 3 2upnext-ff-diasabled | 3 3filters-ff-diasabled

## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] I have considered whether it makes sense to add tests for my changes
- [ ] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...

- [ ] with different themes
- [ ] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/PodcastFragment.kt [hunk 1]
```diff
@@ -611,6 +611,7 @@ class PodcastFragment : BaseFragment(), Toolbar.OnMenuItemClickListener {
                 theme.toggleDarkLightThemeActivity(activity as AppCompatActivity)
                 true
             }
+            it.menu.findItem(R.id.media_route_menu_item).isVisible = !FeatureFlag.isEnabled(Feature.UPNEXT_IN_TAB_BAR)
```
Comment (ashiagr) [inline modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/PodcastFragment.kt:614]: I also removed the cast icon from the Podcast and Cloud screens as the icon is now visible on the mini-player.

**Comment by geekygecko** [comment]
Having the Chromecast icon in the mini player on my Galaxy Z Fold5 is a bit squished.

**Comment by geekygecko** [comment]
Is there a way to disconnect from Chromecast if there is nothing playing and the mini player has disappeared?

**Comment by ashiagr** [comment]
> Is there a way to disconnect from Chromecast if there is nothing playing and the mini player has disappeared?

The cast button exists on the full-screen player. But it isn't visible by default. There was no clear decision regarding this. 
/cc @david-gonzalez-a8c

**Comment by CookieyedCodes** [comment]
@geekygecko @ashiagr @david-gonzalez-a8c

Hmmmmm maby an icon on the artwork and if you tap the artwork (not long press) a quick select popup menu could work?
(Mark as played, Archive, Chromecast, Starred & Share)- Airdrop on iOS

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/PodcastFragment.kt (hunk 1 above)
Comment (ashiagr) [reply modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/PodcastFragment.kt]: I reverted it in https://github.com/Automattic/pocket-casts-android/pull/2101/commits/829cc26f36d6b699367cced67dca540c79e58a26 as I can see the cast icon in iOS app for these screens.

**Comment by ashiagr** [comment]
> Having the Chromecast icon in the mini player on my Galaxy Z Fold5 is a bit squished.

I have tried to make more space for the cast icon in https://github.com/Automattic/pocket-casts-android/pull/2101/commits/9a6c0e337b7ee49ea6e95be78acd31e25601df20. I'll revisit it when I update the mini player design to make it better.  

I'll go ahead and request to review this PR in its current state so I can merge the follow-up approved PRs. I will include an 'empty state' for the mini-player as per the updated designs (pdeCcb-4Xl-p2#comment-4019) in a separate pull request.

**Review by mebarbosa — APPROVED** [review]
Nice work! 🥇 

I smoked test the feature and it worked as before. Also I tested with all themes

**Pull request merged**